ClassPaulita/GuessTheCardGame.cs

```diff
@@ -13,54 +13,58 @@ public class GuessTheCardGame
 
         public GuessTheCardGame()
         {
-            if (File.Exists(_fileName))
+            try
             {
-                StreamReader reader = File.OpenText(_fileName);
-
-                string line = reader.ReadLine();
-
-                while (line != null)
+                if (File.Exists(_fileName))
                 {
-                    string[] scoreAndName = line.Split(' ');
+                    StreamReader reader = File.OpenText(_fileName);
+
+                    string line = reader.ReadLine();
 
-                    if (scoreAndName.Length == 2)
+                    while (line != null)
                     {
-                        bool isItConvertible = int.TryParse(scoreAndName[0], out int result);
+                        string[] scoreAndName = line.Split(' ');
 
-                        if (isItConvertible && scoreAndName[1] != null)
+                        if (scoreAndName.Length == 2)
                         {
-                            _scoreBoard.Add(new KeyValuePair<int, string>(result, scoreAndName[1]));
+                            bool isItConvertible = int.TryParse(scoreAndName[0], out int result);
+
+                            if (isItConvertible && scoreAndName[1] != null)
+                            {
+                                _scoreBoard.Add(new KeyValuePair<int, string>(result, scoreAndName[1]));
+                            }
                         }
+
+                        line = reader.ReadLine();
                     }
 
-                    line = reader.ReadLine();
+                    reader.Close();
                 }
 
-                reader.Close();
+                else
+                {
+                    File.Create(_fileName);
+                }
             }
-
-            else
+            catch (FileNotFoundException exceptionOne)
             {
-                File.Create(_fileName);
+                Console.WriteLine("\nSomething went wrong. Please exit the game and try again.");
             }
-
-            while (_scoreBoard.Count < 5)
+            catch (FileLoadException exceptionTwo)
```

Comment:
Notice, that you are catching two exceptions, but doing the exact same thing with both. Also you are catching specific exceptions, but not using the parameter. Another issue is what if there is a different exception, it won't catch it.

When doing try-catches you always need to have a generic catch to catch everything that could go wrong. If you want to do something specific then that's when you catch a specific exception, i.e. clear up some memory if you get an OutOfMemoryException, but you would still want a generic catch after that catch to catch anything else.

Last thing, since you are catching the exception, it might be helpful to print out the exception's message so the user knows what is wrong.

Good start though, I know this is all new to you.

Reply:
Hey, this is fixed. I just checked it in!